content/en/blog/2022-06-24-transparently-mocking-grpc-services.md

```diff
@@ -0,0 +1,339 @@
+---
+title: "Transparently Mocking gRPC Services"
+slug: "transparently-mocking-grpc-services"
+date: "2022-06-24T12:30:33-05:00"
+draft: true
+image_webp: images/blog/blue_laser.webp
+image: images/blog/blue_laser.jpg
+author: Patrick Deziel
+description: "gRPC is a common framework used to facilitate communication between microservices, but testing these services can be a challenge. In this post we will present a simple strategy for mocking gRPC services in Go."
+---
+
+gRPC is an effective way of implementing service-to-service APIs. However, there are significantly less tools available for mocking and testing than REST-based services. One option is to setup a live test server, however in this blog post we will demonstrate a more lightweight solution which utilizes a hand-built mock. Don't worry, this is easier than it seems!
+<!--more-->
+
+Before we get started, the full tutorial code is available [here](https://github.com/rotationalio/agenda).
+
+## The API
+
+We usually start by defining the API for our service, as it can inform many decisions related to implementation. For simplicity we'll create an "agenda" service in our `.proto` file which contains two RPC definitions and some protocol buffers. The `Schedule` RPC will allow the client to schedule items on the agenda and the `Daily` RPC will return a list of scheduled items to the client.
```

Comment:
minor suggestions on this line

```suggestion
We usually start by defining the API for our service, as it can inform many decisions related to implementation. For simplicity, we'll create an "agenda" service in our `.proto` file, which contains two RPC definitions and some protocol buffers. The `Schedule` RPC will allow the client to schedule items on the agenda and the `Daily` RPC will return a list of scheduled items to the client.
```